knora/dsplib/utils/excel_to_json_resources.py

```diff
@@ -48,7 +48,8 @@ def resources_excel2json(excelfile: str, outfile: str) -> None:
 
     # get overview
     sheet = wb['classes']
-    resource_list = [c for c in sheet.iter_rows(min_row=2, values_only=True)]
+    resource_list = [c for c in sheet.iter_rows(min_row=2, values_only=True)
+                     if any(c) and any([re.search(r'[A-Za-z]+', elem) for elem in c if isinstance(elem, str)])]
```

Comment:
as above: if you just strip each string here, you can just check if it's non-empty, and you don't run the risk of a edge case where the regex misses a valid string

Reply:
It's a bit difficult... You imagine something like the following, I guess?
```
resource_list = [[s.strip() for s in c if isinstance(s, str) else s] for c in sheet.iter_rows(min_row=2, values_only=True)
                     if any(c) and any([re.search(r'[A-Za-z]+', elem) for elem in c if isinstance(elem, str)])]
```

`.strip()` must not be invoked on `None`, so I have to check if s is a string. But if s is not a string, I have to pass it to the method `_extract_row()` anyways, because it requires a fixed number of parameters.
Unfortunately, the above syntax doesn't work, so I have to do it the silly way.
And unfortunately, str(None) == 'None', which is a valid word that matches every regex that looks for words...

Reply:
I'm wondering if the whole thing would be easier if you import the data from an Excel into a pandas DataFrame. If I remember correctly, you added the dependency to pandas some time ago anyway, so you could use it for this part of the code, too? It would allow us to process the data more elegantly. This would involve a bigger refactoring of course.

Reply:
would the syntax work if you write something ternary like `[[s.strip() if isinstance(s, str) else s for s in c] ...`?  
But that's just fooling around with python at this stage :)

Pragmatically I'd say: Yes, pandas could be nice for this. And otherwise, maybe refactor the loading and cleaning of the data into one or two separate methods. Then you can also be a bit more explicit/lengthy about the whole process, so you don't need a one-liner that is so long that it doesn't fit on one line anymore. (Might be a good usecase for a Generator.)